tests/pyspark/test_pyspark_container.py

```diff
@@ -231,3 +233,310 @@ def test_pyspark_nullable():
+                nullable=True,
+            ),
+            T.StructField(
+                name="map",
+                dataType=T.MapType(T.StringType(), T.IntegerType()),
+                nullable=True,
+            ),
+            T.StructField(
+                name="nested_structure",
+                dataType=T.MapType(
+                    T.ArrayType(T.StringType()),
+                    T.MapType(T.StringType(), T.ArrayType(T.StringType())),
+                ),
+                nullable=True,
+            ),
+        ]
+    )
+
+
+def test_schema_to_ddl(schema_with_complex_datatypes):
+    """
+    Test the conversion from a schema to a DDL string through `to_ddl()`.
+    """
+
+    assert schema_with_complex_datatypes.to_ddl() == ",".join(
+        [
+            "non_nullable INT",
+            "binary BINARY",
+            "byte TINYINT",
+            "text STRING",
+            "integer INT",
+            "long BIGINT",
+            "float FLOAT",
+            "double DOUBLE",
+            "boolean BOOLEAN",
+            "decimal DECIMAL(10,0)",
+            "date DATE",
+            "timestamp TIMESTAMP",
+            "timestamp_ntz TIMESTAMP",
+            "array ARRAY<STRING>",
+            "map MAP<STRING, INT>",
+            "nested_structure MAP<ARRAY<STRING>, MAP<STRING, ARRAY<STRING>>>",
+        ]
+    )
+    assert schema_with_complex_datatypes.to_ddl() != ",".join(
+        [
+            "non_nullable INT",
+            "binary STRING",  # Wrong
+            "byte STRING",  # Wrong
+            "text STRING",
+            "integer INT",
+            "long BIGINT",
+            "float FLOAT",
+            "double DOUBLE",
+            "boolean BOOLEAN",
+            "decimal DECIMAL(10,0)",
+            "date DATE",
+            "timestamp TIMESTAMP",
+            "timestamp_ntz TIMESTAMP",
+            "array ARRAY<STRING>",
+            "map MAP<STRING, INT>",
+            "nested_structure MAP<ARRAY<STRING>, MAP<STRING, ARRAY<STRING>>>",
+        ]
+    )
+
+
+@pytest.fixture(scope="module")
+def schema_with_simple_datatypes():
+    """
+    Model containing all common datatypes for PySpark namespace, supported by CSV.
+    """
+    schema = DataFrameSchema(
+        {
+            "non_nullable": Column(T.IntegerType(), nullable=False),
+            "byte": Column(T.ByteType()),
+            "text": Column(T.StringType()),
+            "integer": Column(T.IntegerType()),
+            "long": Column(T.LongType()),
+            "float": Column(T.FloatType()),
+            "double": Column(T.DoubleType()),
+            "boolean": Column(T.BooleanType()),
+            "decimal": Column(T.DecimalType()),
+            "date": Column(T.DateType()),
+            "timestamp": Column(T.TimestampType()),
+            "timestamp_ntz": Column(T.TimestampNTZType()),
+        }
+    )
+
+    return schema
+
+
+def test_pyspark_read(schema_with_simple_datatypes, tmp_path, spark):
```

Comment:
@filipeo2-mck can we mark this as skipped if the os is windows? Looks like we do this [elsewhere in the tests](https://github.com/unionai-oss/pandera/blob/11488670c11a09e605f251b5d62bd0a88a57672c/tests/io/test_io.py#L729)

```python
@pytest.mark.skipif(
    platform.system() == "Windows",
    reason="skipping due to issues with opening file names for temp files.",
)
```

Reply:
Done! Thank you for checking it. The Windows jobs are running now.